Model/CodedAttributeOptionRepository.php

```diff
@@ -54,14 +59,33 @@ public function delete($entityType, $attributeCode, $optionCode)
 
     private function addOption($entityType, $attributeCode, $optionCode, MagentoAttributeOption $option)
     {
-        $maxExistingOptionId = $this->optionCodeRepository->getMaxOptionId($entityType, $attributeCode) ?: 0;
-        $this->optionManagementService->add($entityType, $attributeCode, $option);
-        $newOptionId = $this->optionCodeRepository->getMaxOptionId($entityType, $attributeCode) ?: 0;
+        $newOptionId = $this->optionManagementService->add($entityType, $attributeCode, $option);
 
-        if ($newOptionId <= $maxExistingOptionId) {
-            throw new \RuntimeException('Failed to add option.');
-        }
+        // Magento returns the new option id prefixed with "id_"
+        // We remove non numeric characters to get the correct option id
+        $newOptionId = preg_replace("/[^0-9]/", "", $newOptionId);
 
         $this->optionCodeRepository->setOptionId($entityType, $attributeCode, $optionCode, $newOptionId);
     }
+
+    public function updateOption($attributeCode, MagentoAttributeOption $magentoOption)
+    {
+        $attribute = $this->productAttributeRepository->get($attributeCode);
+        $options = $attribute->getOptions();
+        foreach ($options as $option) {
+            /** @var MagentoAttributeOption $option */
+            if ($option->getValue() == $magentoOption->getValue()) {
```

Comment:
This is a string comparison yeah? IMO it should use a string comparison function or a strict `===` otherwise you can get some truthy falsey problems like follow.

Tbh i don't expect this to ever be an issue with these LOC, but it's the only thing i've spotted here :) 

```bash
$ cat lol.php
<?php
$left = 'true';
$right = true;
var_dump($left == $right);

$ php lol.php
/Users/lukerodgers/lol.php:4:
bool(true)
```